process/transaction/shardProcess.go

```diff
@@ -99,6 +115,12 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) error
 
 	err = txProc.checkTxValues(tx, acntSnd)
 	if err != nil {
+		if err == process.ErrInsufficientFunds {
+			receiptErr := txProc.createReceiptsWhenFail(tx, acntSnd)
```

Comment:
Why not pass `err` to `txProc.createReceiptsWhenFail()`, so that it is added to the Data field of the Receipt?

Reply:
done.